user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/MeasurementMain.java

```diff
@@ -418,33 +403,62 @@ private void handleDeletionResults(Result<Void, MeasurementDeletionException> re
     result.onValue(v -> measurementDetailsComponent.refreshGrids());
   }
 
-  private void executeInUi(Command command) {
-    getUI().ifPresent(ui -> ui.access(command));
-  }
-
   private void downloadProteomicsMetadata(List<String> selectedMeasurementIds) {
+    UI ui = getUI().orElse(null);
+    if (ui == null) {
+      return; // UI has been detached
+    }
     ProjectId projectId = context.projectId().orElseThrow();
     var inProgressToast = messageFactory.pendingTaskToast("measurement.preparing-download",
         MessageSourceNotificationFactory.EMPTY_PARAMETERS, getLocale());
+
+    if (ui.getSession() != null && ui.getSession().hasLock()) {
+      inProgressToast.open();
+    } else {
+      ui.access(inProgressToast::open);
+    }
```

Comment:
you show the inprogress toast efore the async service started work. The reactor publisher is not active at this point in time. 
Is this intended?